extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```
@@ -70,10 +73,17 @@
     CamelRuntimeBuildItem create(
             CamelRecorder recorder,
             List<CamelRegistryBuildItem> registryItems,
+            List<CamelContextBuildItem> contexts,
             BuildProducer<RuntimeBeanBuildItem> runtimeBeans) {
 
+        Supplier<CamelContext> contextSupplier;
+        if (contexts.size() == 0) {
+            contextSupplier = new FastCamelContextSupplier();
+        } else {
+            contextSupplier = contexts.get(0).getContext();
+        }
```

We may need to throw an exception in case of contexts > 1